molo/core/models.py

```diff
@@ -12,9 +14,31 @@
 from wagtail.wagtailimages.blocks import ImageChooserBlock
 
 from molo.core.blocks import MarkDownBlock
-
-
-class HomePage(Page):
+from molo.core import constants
+
+
+class CommentedPageMixin(object):
+    def get_effective_commenting_settings(self):
+        # return commenting settings for the homepage
+        if self.commenting_state:
+            return {
+                'state': self.commenting_state,
+                'open_time': self.commenting_open_time,
+                'close_time': self.commenting_close_time
+            }
+        # use the commenting settings for the parent page
+        parent_page = Page.objects.all().ancestor_of(self).last().specific
+        if (parent_page):
+            return parent_page.get_effective_commenting_settings()
+        # add a default in case nothing is set. We should never get here
+        return {
+            'state': constants.COMMENTING_DISABLED,
+            'open_time': None,
+            'close_time': None
+        }
+
+
+class HomePage(CommentedPageMixin, Page):
```

Comment:
The full story of how the list of classes to check is linearized is very complicated, by thankfully Python classes have a funky `__mro__` attribute that makes answering the question in practice a bit easier, e.g.
```python
>>> class A(object):
...    pass
... 
>>> class B(object):
...    pass
... 
>>> class C(A, B):
...    pass
... 
>>> C.__mro__
(<class '__main__.C'>, <class '__main__.A'>, <class '__main__.B'>, <type 'object'>)
```
So if you want to check the mixin first, and then `Page`, then `Page` should be on the right as @KaitCrawford has it. The point of mixins though is that they shouldn't override any of the base classes methods and that order shouldn't matter too much (i.e. mixins should aim to keep things simple).